server/modules/survey/service/export/jobs/PrepareData.js

```diff
@@ -0,0 +1,39 @@
+import * as Survey from '@core/survey/survey'
+import * as ProcessUtils from '@core/processUtils'
+import * as DateUtils from '@core/dateUtils'
+
+import Job from '@server/job/job'
+import * as FileUtils from '@server/utils/file/fileUtils'
+import * as SurveyManager from '@server/modules/survey/manager/surveyManager'
+import * as SurveyRdbManager from '@server/modules/surveyRdb/manager/surveyRdbManager'
+import * as CSVWriter from '@server/utils/file/csvWriter'
+
+export default class PrepareData extends Job {
+  constructor(params) {
+    super('PrepareData', params)
+  }
+
+  async execute() {
+    const { surveyId } = this.context
+
+    const survey = await SurveyManager.fetchSurveyById(surveyId, true, false, this.tx)
+
+    const surveyInfo = Survey.getSurveyInfo(survey)
+    const surveyName = Survey.getName(surveyInfo)
+
+    const exportDataFolderName = `${surveyName}_export_${DateUtils.nowFormatDefault()}`
+    const dir = FileUtils.join(ProcessUtils.ENV.tempFolder, exportDataFolderName)
+    await FileUtils.rmdir(dir)
+    await FileUtils.mkdir(dir)
+
+    const tables = await SurveyRdbManager.getTables(surveyId, this.tx)
+    await Promise.all(
+      tables.map(async ({ schemaname, tablename }) => {
+        const data = await this.tx.query(`SELECT * from ${schemaname}.${tablename}`)
```

Comment:
you need to use a manager to get the data

Reply:
a stream is more efficient (see data download from explorer )

Reply:
composite attributes should have separate columns (see explorer)

Reply:
only attributes must be exported, (attached pictures show columns must not be included)
<img width="816" alt="Screenshot 2021-03-25 at 07 29 05" src="https://user-images.githubusercontent.com/1138156/112428696-d91bd980-8d3b-11eb-8a2d-8239ce30c42d.png">